optimade/server/config.py

```diff
@@ -51,6 +57,12 @@ class ServerConfig(BaseSettings):
     use_real_mongo: bool = Field(
         False, description="Use a real Mongo server rather than MongoMock"
     )
+
+    database_backend: SupportedBackend = Field(
```

Comment:
What if the config would take a qualified class name for the entry collection implementation (instead or in addition to the backend enum). This could allow to inject a custom implementation in cases were you need to alter the OPT implementations? Related to #657.

Reply:
Again, this is something we discussed (see OP second to-do list); definitely the way to go (I think @shyamd even volunteered...) but it can wait for a future PR.